src/app/services/xml-parsers/msdesc-parser.ts

```diff
@@ -0,0 +1,954 @@
+import { flat, getBoolean } from 'src/app/utils/js-utils';
+import {
+    AccMat, Acquisition, Additional, Additions, AdminInfo, AltIdentifier, Binding, BindingDesc, Collation, CollectionEl, Condition,
+    CustEvent, CustodialHist, DecoDesc, DecoNote, Depth, Dim, Dimensions, Explicit, Filiation, FinalRubric, Foliation,
+    G,
+    HandDesc, Head, Height, History, Identifier, Incipit, Institution, Layout, LayoutDesc, Locus, LocusGrp, MaterialValues, MsContents,
+    MsDesc, MsFrag, MsIdentifier, MsItem, MsItemStruct, MsName, MsPart, MusicNotation, Note, ObjectDesc, OrigDate,
+    Origin, OrigPlace, Paragraph, PhysDesc, Provenance, RecordHist, Repository, Rubric, ScriptDesc, Seal, SealDesc, Source, Summary,
+    Support, SupportDesc, Surrogates, TypeDesc, TypeNote, Width, XMLElement,
+} from '../../models/evt-models';
+import { GapParser, GenericElemParser, LBParser, NoteParser, ParagraphParser, queryAndParseElement, queryAndParseElements } from './basic-parsers';
+import { GParser } from './character-declarations-parser';
+import { createParser, getClass, getDefaultN, getID, parseChildren, Parser } from './parser-models';
+
+function unhandledElement(xml: XMLElement, name: string) {
+    return flat(Array.from(xml.querySelectorAll<XMLElement>(`:scope > ${name}`)).map(e => parseChildren(e, this.genericParse)));
+}
```

Comment:
@giuliac89 I would put this in `parser-model.ts` file since it could be useful also for other parsers. What do you think?